src/scripts/main.js

```diff
@@ -1,5 +1,27 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  let x = e.clientX;
+  let y = e.clientY;
+  const spider = document.querySelector('.spider');
+  const spiderCoords = spider.getBoundingClientRect();
+  const wall = document.querySelector('.wall');
+  const wallCoords = wall.getBoundingClientRect();
+
+  x -= spiderCoords.width / 2;
+  y -= spiderCoords.height / 2;
+
+  x = Math.max(
+    wallCoords.left,
+    Math.min(x, wallCoords.right - spiderCoords.width),
+  );
+
+  y = Math.max(
+    wallCoords.top,
+    Math.min(y, wallCoords.bottom - spiderCoords.height),
+  );
+
+  spider.style.position = 'fixed';
```

Comment:
Setting the position to 'fixed' might not be necessary if the spider element is already positioned correctly. Ensure this is intended, as it will remove the element from the normal document flow.